src/ui/repo-tree-model.cpp

```diff
@@ -61,8 +61,11 @@ RepoTreeModel::RepoTreeModel(QObject *parent)
 }
 RepoTreeModel::~RepoTreeModel()
 {
-    if (item(kIndexOfVirtualReposCategory) != virtual_repos_category_)
-        delete virtual_repos_category_;
+    // if we miss it, just add it and
+    // let dtor of repotreemodel to delete it automatically
+    if (item(kIndexOfVirtualReposCategory) != virtual_repos_category_) {
+        insertRow(kIndexOfVirtualReposCategory, virtual_repos_category_);
+    }
```

Comment:
When a object is being destroyed:

1. the code of its destructor(or dtor, in short)'s function body is executed
2. (after 1) the code at the end of function, i.e. `}`, is executed. it does destruction of all elements of the object.

So, in this case, the inserting code is executed before the the real part of dtor is executed.

Feel free if you have any questions